PlayerInteractorController and Interaction Objects

meghanai28 opened the pull request:
Coded playerinteractioncontroller and interaction objects for game.

meghanai28 committed:
PlayerInteractorController and Interaction Objects

Coded playerinteractioncontroller and interaction objects for game.

getmikyled commented on CannibalisticZombies/.vscode/settings.json:
```diff
@@ -0,0 +1,56 @@
+{
```
Do you know what these changes are from? 
In the future if there are changes you don't recognize doing, lets leave it on your local side and not have it committed pushed to the branch

getmikyled commented on CannibalisticZombies/Assets/Scripts/Door.cs:
```diff
@@ -0,0 +1,75 @@
+                keyBindText.text = "Press E to Open Door";
+            }
+            /// show the UI on the screen
+            keyBindUI.SetActive(true);
```
I noticed that we call this here and in InteractionObject!

C#'s version of `super()` is `base`!

In this case I think it would be `base.OnPointerEnter(eventData);`
(I may be wrong tho because personally I have never had to use it so feel free to double check)

getmikyled commented on CannibalisticZombies/Assets/Scripts/InteractionObject.cs:
```diff
@@ -0,0 +1,59 @@
+    public class InteractionObject : MonoBehaviour, IPointerEnterHandler, IPointerExitHandler
+    {
+        ///  This is a variable that is used to grab the canvas associated with the keyBind UI
+        [SerializeField] public GameObject keyBindUI;
```
From what I have learned, using `[SerializeField]` and `public` is redundant since both expose the variable into the inspector. Since we won't need to access keyBindUI from other classes tho, I think it is okay to declare it as private instead and keep `[SerializeField]`

getmikyled commented on CannibalisticZombies/Assets/Scripts/PickupInteractionObject.cs:
```diff
@@ -0,0 +1,54 @@
+        public override void OnPointerEnter(PointerEventData eventData)
+        {
+            /// grab the textComponent from the keyBindUI, which is a child (GetComponentInChildren) of the canvas that seralizes the keyBind Field.
+            Text keyBindText = keyBindUI.GetComponentInChildren<Text>();
```
Does every object in the scene have their own keybind text? If so we can just set the text in the editor since it will be the same per object instead of coding it manually.

getmikyled commented on CannibalisticZombies/ProjectSettings/TagManager.asset:
```diff
@@ -11,7 +11,7 @@ TagManager:
   - Water
   - UI
-  - 
+  - interactable
```
Correct me if I am wrong, but do you use interactable to identify objects in the code? If so I cannot find it. Since we do not use it, I'd say we can live without the interactable tag for now.

getmikyled reviewed (commented):
Done leaving feedback!

meghanai28 committed:
Minor Changes Based On Feedback

Added some Setters to interactionObject, and improved usage of inhertance and serializefield

meghanai28 committed:
Delete CannibalisticZombies/.vscode/settings.json

getmikyled commented:
Looks good! Merging it in!
